bom_stock/models/product_product.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+# © 2015 Camptocamp SA, Sodexis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, fields, api
+from openerp.addons import decimal_precision as dp
+
+
+class Product(models.Model):
+    _inherit = 'product.product'
+
+    @api.multi
+    def _bom_stock_mapping(self):
+        return {'real': 'qty_available',
+                'virtual': 'virtual_available',
+                'immediately': 'immediately_usable_qty'}
+
+    @api.one
+    def _compute_bom_stock(self):
```

Comment:
As this is not stored, the performance will suffer a lot. You should set correctly field dependencies, and make it stored.